centinel/primitives/http.py

```diff
@@ -49,10 +52,64 @@ def get_request(host, path="/", headers=None, ssl=False,
     }
     # the external result is used when threading to store
     # the results in the list container provided.
-    if external is not None and type(external) is dict:
-        external[url] = result
+    # if external is not None and type(external) is dict:
+    #    external[url] = result
     return result
 
+
+def get_request(host, path="/", headers=None, ssl=False,
+                external=None, url=None):
+    http_results = {}
+    first_response = _get_http_request(host, path, headers, ssl, external, url)
+    if "failure" in first_response["response"]:  # If there was an error, just ignore redirects and return
+        if external is not None and type(external) is dict:
+            external[url] = first_response
+            return first_response
+    # Checks HTTP Status code and location header to see if the webpage calls for a redirect
+    is_redirecting = str(first_response["response"]["status"]).startswith("3") and "location"\
+                                            in first_response["response"]["headers"]
+    if is_redirecting:
+        http_results["request"] = first_response["request"]
+        http_results["redirects"] = {}
+        http_results["redirects"]["0 (" + host + path + ")"] = first_response["response"]
```

Comment:
if you intend to do indexing, it might be better to make it easier to access the host, path, and order (e.g. 0). You could do something like an array of dicts, where the index is the order and include a host and path field in the dict.